feat: Create proper results for CustomInterface

hlomzik opened the pull request:
First try

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

hlomzik committed:
feat: Create proper results for CustomInterface

First try

Commit:
Fix import path

niklub commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18078010277)
> Successfully merged:  create mode 100644 label_studio/tests/sdk/test_project_annotators_api.py

niklub committed:
Merge branch 'develop' into 'fb-custom-regions'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18078010277

hlomzik commented on web/libs/editor/src/tags/custom/Custom.jsx:
```diff
@@ -126,16 +126,6 @@ const Model = types
-      try {
-        const areaValue = { custominterface: value };
-        const resultValue = { custominterface: value };
-        self.annotation.createResult(areaValue, resultValue, self, self.toname);
```
why did you remove the main call creating result and region?

hlomzik commented on web/libs/editor/src/tags/custom/Custom.jsx:
```diff
@@ -249,12 +239,18 @@ const Model = types
+            try {
+              const res = store.events.invoke("submitAnnotation", store, annotation);
+              if (res && typeof res.catch === "function") res.catch(() => {});
+            } catch (e) {}
```
we have to remove all these calls completely! result update should not submit annotation, that's totally wrong. also it looks like it doesn't work anyway.

hlomzik commented on web/libs/editor/src/tags/custom/index.ts:
```diff
@@ -11,7 +11,9 @@ Registry.addCustomTag("CustomInterface", {
-  detector: (sn: any) => Boolean(sn?.value?.custominterface || sn?.custominterface),
+  // Do not detect regions by value for CustomInterface. Classification areas (global results)
+  // should deserialize as ClassificationArea, not CustomRegionModel, otherwise addResult is missing.
+  detector: () => false,
```
if there are problems with false detection of classification we should fix the problem, not break the detection completely.

hlomzik committed:
Properly fix issues with CustomRegion

- Proper relations to other parts of LSF like `object` and types
- Remove some excess functions and fields breaking the LSF
- Add missing AreaMixin

hlomzik committed:
Remove HtxCustomRegionView, it's useless

hlomzik committed:
Remove weird and dangerous code

We should not call annotation submission methods from the tag,
it's totally wrong.

hlomzik committed:
Remove wrong export

hlomzik committed:
Multiple important fixes

- Add missing serialize() to save results properly
- Add `update(value)` and `get value` for better region api
- Fix tag type: not a control, not a classification; we should use ObjectBase
- Remove more useless code
- Use `observer()` for rendered tag so it reacts on region changes
  - Also fix dependency to react on regions

hlomzik committed:
Also accept CDATA content

That allows to use usual JS code with &, <, etc.

hlomzik committed:
Also update result along with the region
